src/core/border_router/routing_manager.cpp

```diff
@@ -406,98 +406,76 @@ void RoutingManager::UpdateDiscoveredPrefixTableOnNetDataChange(void)
     mDiscoveredPrefixTable.SetAllowDefaultRouteInNetData(foundDefRouteOmrPrefix);
 }
 
-void RoutingManager::EvaluateOmrPrefix(OmrPrefixArray &aNewOmrPrefixes)
+void RoutingManager::EvaluateOmrPrefix(OnMeshPrefixArray &aNewPrefixes)
 {
     NetworkData::Iterator           iterator = NetworkData::kIteratorInit;
     NetworkData::OnMeshPrefixConfig onMeshPrefixConfig;
-    OmrPrefix *                     favoredOmrEntry = nullptr;
-    OmrPrefix *                     localOmrEntry   = nullptr;
+    OmrPrefix                       favoredOmrPrefix;
 
     OT_ASSERT(mIsRunning);
 
     while (Get<NetworkData::Leader>().GetNextOnMeshPrefix(iterator, onMeshPrefixConfig) == kErrorNone)
     {
-        OmrPrefix *entry;
-
-        if (!IsValidOmrPrefix(onMeshPrefixConfig))
+        if (!onMeshPrefixConfig.mOnMesh)
```

Comment:
Should we ignore Domain Prefix since DUAs are ND proxied, thus should not be routable at the same time.

Reply:
Thanks. Sounds good to me. Will change it in next push.

Reply:
Correct, any Domain Prefix is advertised as on-link on the AIL / backbone link. Therefore it shouldn't be advertised as a route.